### ISSUE TITLE
One of our projects fails to build with: "csc.exe" exited with code -2146232797

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#33542]
I don't have any idea why this would happen, this used to build on everything except Visual Studio 2019 Preview 2, preview 1.1 worked.

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/434543/one-of-our-projects-fails-to-build-with-cscexe-exi.html
VSTS ticketId: 776735_
_These are the original issue comments:_

tstedel on 1/31/2019, 05:49 PM (19 days ago): <p>Still does this in preview 2.1. It hangs for a very long time and then doesn't build.</p><p>1&gt;------ Build started: Project: POS_Connector.Core, Configuration: Debug Any CPU ------<br>1&gt;C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\15.0\Bin\Roslyn\Microsoft.CSharp.Core.targets(52,5): error MSB6006: "csc.exe" exited with code -2146232797.<br>========== Build: 0 succeeded, 1 failed, 3 up-to-date, 0 skipped ==========</p><p><strong></strong><u></u><sub></sub><sup></sup></p>

tstedel on 1/31/2019, 06:41 PM (19 days ago): <pre>using System.Threading.Tasks;
namespace ConsoleApp6
\{
    class Program
    \{
        static async Task Main(string[] args)
        \{
            try
            \{
                try
                \{
                    return;
                }
                finally
                \{
                    await Task.CompletedTask;
                }
            }
            catch \{ }
            finally
            \{
                await Task.CompletedTask;
            }
        }
    }
}</pre>

tstedel on 1/31/2019, 06:42 PM (19 days ago): <p>What I just posted is the minimum necessary to reproduce this issue</p>

_These are the original issue solutions:_
(no solutions)

**Comment by sharwell** [comment]
@jaredpar I believe this is COR_E_FAILFAST

**Comment by RikkiGibson** [comment]
Looks like this builds in dev15 but not dev16.

**Comment by RikkiGibson** [comment]
I have a reproducer in a branch. Check out the stack trace.

```
Microsoft.CodeAnalysis.dll!Roslyn.Utilities.ExceptionUtilities.UnexpectedValue(object o) Line 18	C#
Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.CodeGen.ILOpCodeExtensions.GetLeaveOpcode(System.Reflection.Metadata.ILOpCode opcode) Line 38	C#
Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.CodeGen.ILBuilder.BasicBlock.RewriteBranchesAcrossExceptionHandlers() Line 294	C#
Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.CodeGen.ILBuilder.RewriteBranchesAcrossExceptionHandlers() Line 761	C#
Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.CodeGen.ILBuilder.RealizeBlocks() Line 853	C#
Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.CodeGen.ILBuilder.Realize() Line 189	C#
Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.GenerateImpl() Line 270	C#
Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.Generate(out int asyncCatchHandlerOffset, out System.Collections.Immutable.ImmutableArray<int> asyncYieldPoints, out System.Collections.Immutable.ImmutableArray<int> asyncResumePoints) Line 202	C#
Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.MethodCompiler.GenerateMethodBody(Microsoft.CodeAnalysis.CSharp.Emit.PEModuleBuilder moduleBuilder, Microsoft.CodeAnalysis.CSharp.Symbols.MethodSymbol method, int methodOrdinal, Microsoft.CodeAnalysis.CSharp.BoundStatement block, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.CodeGen.LambdaDebugInfo> lambdaDebugInfo, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.CodeGen.ClosureDebugInfo> closureDebugInfo, Microsoft.CodeAnalysis.CSharp.StateMachineTypeSymbol stateMachineTypeOpt, Microsoft.CodeAnalysis.CodeGen.VariableSlotAllocator variableSlotAllocatorOpt, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, Microsoft.CodeAnalysis.CodeGen.DebugDocumentProvider debugDocumentProvider, Microsoft.CodeAnalysis.CSharp.ImportChain importChainOpt, bool emittingPdb, bool emitTestCoverageData, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.CodeGen.SourceSpan> dynamicAnalysisSpans) Line 1409	C#
Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileSynthesizedMethods(Microsoft.CodeAnalysis.CSharp.TypeCompilationState compilationState) Line 696	C#
Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamedType(Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol containingType) Line 602	C#
Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.MethodCompiler.VisitNamedType(Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol symbol, Microsoft.CodeAnalysis.CSharp.TypeCompilationState arg) Line 376	C#
Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol.Accept<Microsoft.CodeAnalysis.CSharp.TypeCompilationState, object>(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor<Microsoft.CodeAnalysis.CSharp.TypeCompilationState, object> visitor, Microsoft.CodeAnalysis.CSharp.TypeCompilationState argument) Line 576	C#
Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamespace(Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol symbol) Line 355	C#
Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.MethodCompiler.VisitNamespace(Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol symbol, Microsoft.CodeAnalysis.CSharp.TypeCompilationState arg) Line 330	C#
Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol.Accept<Microsoft.CodeAnalysis.CSharp.TypeCompilationState, object>(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor<Microsoft.CodeAnalysis.CSharp.TypeCompilationState, object> visitor, Microsoft.CodeAnalysis.CSharp.TypeCompilationState argument) Line 135	C#
Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamespace(Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol symbol) Line 355	C#
Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileMethodBodies(Microsoft.CodeAnalysis.CSharp.CSharpCompilation compilation, Microsoft.CodeAnalysis.CSharp.Emit.PEModuleBuilder moduleBeingBuiltOpt, bool emittingPdb, bool emitTestCoverageData, bool hasDeclarationErrors, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, System.Predicate<Microsoft.CodeAnalysis.CSharp.Symbol> filterOpt, System.Threading.CancellationToken cancellationToken) Line 143	C#
Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpCompilation.CompileMethods(Microsoft.CodeAnalysis.Emit.CommonPEModuleBuilder moduleBuilder, bool emittingPdb, bool emitMetadataOnly, bool emitTestCoverageData, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, System.Predicate<Microsoft.CodeAnalysis.ISymbol> filterOpt, System.Threading.CancellationToken cancellationToken) Line 2626	C#
Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Compilation.Emit(System.IO.Stream peStream, System.IO.Stream metadataPEStream, System.IO.Stream pdbStream, System.IO.Stream xmlDocumentationStream, System.IO.Stream win32Resources, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.ResourceDescription> manifestResources, Microsoft.CodeAnalysis.Emit.EmitOptions options, Microsoft.CodeAnalysis.IMethodSymbol debugEntryPoint, System.IO.Stream sourceLinkStream, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.EmbeddedText> embeddedTexts, Microsoft.CodeAnalysis.CodeGen.CompilationTestData testData, System.Threading.CancellationToken cancellationToken) Line 2405	C#
Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Compilation.Emit(System.IO.Stream peStream, System.IO.Stream pdbStream, System.IO.Stream xmlDocumentationStream, System.IO.Stream win32Resources, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.ResourceDescription> manifestResources, Microsoft.CodeAnalysis.Emit.EmitOptions options, Microsoft.CodeAnalysis.IMethodSymbol debugEntryPoint, System.IO.Stream sourceLinkStream, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.EmbeddedText> embeddedTexts, System.IO.Stream metadataPEStream, System.Threading.CancellationToken cancellationToken) Line 2349	C#
Roslyn.Test.Utilities.dll!Microsoft.CodeAnalysis.DiagnosticExtensions.GetEmitDiagnostics<Microsoft.CodeAnalysis.CSharp.CSharpCompilation>(Microsoft.CodeAnalysis.CSharp.CSharpCompilation c, Microsoft.CodeAnalysis.Emit.EmitOptions options, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.ResourceDescription> manifestResources) Line 276	C#
Roslyn.Test.Utilities.dll!Microsoft.CodeAnalysis.DiagnosticExtensions.VerifyEmitDiagnostics<Microsoft.CodeAnalysis.CSharp.CSharpCompilation>(Microsoft.CodeAnalysis.CSharp.CSharpCompilation c, Microsoft.CodeAnalysis.Emit.EmitOptions options, Microsoft.CodeAnalysis.Test.Utilities.DiagnosticDescription[] expected) Line 265	C#
Roslyn.Test.Utilities.dll!Microsoft.CodeAnalysis.DiagnosticExtensions.VerifyEmitDiagnostics<Microsoft.CodeAnalysis.CSharp.CSharpCompilation>(Microsoft.CodeAnalysis.CSharp.CSharpCompilation c, Microsoft.CodeAnalysis.Test.Utilities.DiagnosticDescription[] expected) Line 282	C#
Microsoft.CodeAnalysis.CSharp.Emit.UnitTests.dll!Microsoft.CodeAnalysis.CSharp.UnitTests.EmitErrorTests.Repro_33542() Line 393	C#
```

**Comment by JohnGalt1717** [comment]
I'm getting this with a dot.net core 2.2 (SDK 105) project.  Just yesterday it started failing in both the latest version of VS.net 2017 and the RTM of 2019 if it matters.

**Comment by jaredpar** [comment]
@JohnGalt1717 is there some source you could share out that would help us understand if this is related to other issues we're looking into?

**Comment by JohnGalt1717** [comment]
@jaredpar Unfortunately not. One project doesn't work, the other does.

There appears to be something really wonky going on with .NET core 2.2. sdks though.  VS.net 2019 installs 2.2.202 but the most recent listed on dot.net is 2.2.105 which makes no sense at all and I think is part of the problem.

Using 2.2.105 in our global.json file using FROM mcr.microsoft.com/dotnet/core/sdk:2.2 AS build in our docker results in the following error now as well which doesn't happen at command prompt in windows:

`/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error : FailFast: [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error : System.InvalidOperationException: Unexpected value 'Beq' of type 'System.Reflection.Metadata.ILOpCode' [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CodeGen.ILOpCodeExtensions.GetLeaveOpcode(ILOpCode opcode) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CodeGen.ILBuilder.BasicBlock.RewriteBranchesAcrossExceptionHandlers() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CodeGen.ILBuilder.RewriteBranchesAcrossExceptionHandlers() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CodeGen.ILBuilder.RealizeBlocks() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CodeGen.ILBuilder.Realize() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.GenerateImpl() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.Generate(Int32& asyncCatchHandlerOffset, ImmutableArray`1& asyncYieldPoints, ImmutableArray`1& asyncResumePoints) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.MethodCompiler.GenerateMethodBody(PEModuleBuilder moduleBuilder, MethodSymbol method, Int32 methodOrdinal, BoundStatement block, ImmutableArray`1 lambdaDebugInfo, ImmutableArray`1 closureDebugInfo, StateMachineTypeSymbol stateMachineTypeOpt, VariableSlotAllocator variableSlotAllocatorOpt, DiagnosticBag diagnostics, DebugDocumentProvider debugDocumentProvider, ImportChain importChainOpt, Boolean emittingPdb, Boolean emitTestCoverageData, ImmutableArray`1 dynamicAnalysisSpans) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileSynthesizedMethods(TypeCompilationState compilationState) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamedType(NamedTypeSymbol containingType) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.MethodCompiler.<>c__DisplayClass21_0.<CompileNamedTypeAsTask>b__0() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :  [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at System.Environment.FailFast(System.String, System.Exception) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at System.Environment.FailFast(System.String, System.Exception) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.FailFast.OnFatalException(System.Exception) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.FatalError.Report(System.Exception, System.Action`1<System.Exception>) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.FatalError.ReportUnlessCanceled(System.Exception) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.MethodCompiler+<>c__DisplayClass21_0.<CompileNamedTypeAsTask>b__0() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CodeGen.ILOpCodeExtensions.GetLeaveOpcode(System.Reflection.Metadata.ILOpCode) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CodeGen.ILBuilder+BasicBlock.RewriteBranchesAcrossExceptionHandlers() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CodeGen.ILBuilder.RewriteBranchesAcrossExceptionHandlers() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CodeGen.ILBuilder.RealizeBlocks() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CodeGen.ILBuilder.Realize() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.GenerateImpl() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.Generate(Int32 ByRef, System.Collections.Immutable.ImmutableArray`1<Int32> ByRef, System.Collections.Immutable.ImmutableArray`1<Int32> ByRef) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.MethodCompiler.GenerateMethodBody(Microsoft.CodeAnalysis.CSharp.Emit.PEModuleBuilder, Microsoft.CodeAnalysis.CSharp.Symbols.MethodSymbol, Int32, Microsoft.CodeAnalysis.CSharp.BoundStatement, System.Collections.Immutable.ImmutableArray`1<Microsoft.CodeAnalysis.CodeGen.LambdaDebugInfo>, System.Collections.Immutable.ImmutableArray`1<Microsoft.CodeAnalysis.CodeGen.ClosureDebugInfo>, Microsoft.CodeAnalysis.CSharp.StateMachineTypeSymbol, Microsoft.CodeAnalysis.CodeGen.VariableSlotAllocator, Microsoft.CodeAnalysis.DiagnosticBag, Microsoft.CodeAnalysis.CodeGen.DebugDocumentProvider, Microsoft.CodeAnalysis.CSharp.ImportChain, Boolean, Boolean, System.Collections.Immutable.ImmutableArray`1<Microsoft.CodeAnalysis.CodeGen.SourceSpan>) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileSynthesizedMethods(Microsoft.CodeAnalysis.CSharp.TypeCompilationState) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamedType(Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.MethodCompiler+<>c__DisplayClass21_0.<CompileNamedTypeAsTask>b__0() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Roslyn.Utilities.UICultureUtilities+<>c__DisplayClass5_0.<WithCurrentUICulture>b__0() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at System.Threading.ThreadPoolWorkQueue.Dispatch() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :  [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error : Exception details: [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error : System.InvalidOperationException: Unexpected value 'Beq' of type 'System.Reflection.Metadata.ILOpCode' [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CodeGen.ILOpCodeExtensions.GetLeaveOpcode(ILOpCode opcode) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CodeGen.ILBuilder.BasicBlock.RewriteBranchesAcrossExceptionHandlers() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CodeGen.ILBuilder.RewriteBranchesAcrossExceptionHandlers() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CodeGen.ILBuilder.RealizeBlocks() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CodeGen.ILBuilder.Realize() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.GenerateImpl() [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.Generate(Int32& asyncCatchHandlerOffset, ImmutableArray`1& asyncYieldPoints, ImmutableArray`1& asyncResumePoints) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.MethodCompiler.GenerateMethodBody(PEModuleBuilder moduleBuilder, MethodSymbol method, Int32 methodOrdinal, BoundStatement block, ImmutableArray`1 lambdaDebugInfo, ImmutableArray`1 closureDebugInfo, StateMachineTypeSymbol stateMachineTypeOpt, VariableSlotAllocator variableSlotAllocatorOpt, DiagnosticBag diagnostics, DebugDocumentProvider debugDocumentProvider, ImportChain importChainOpt, Boolean emittingPdb, Boolean emitTestCoverageData, ImmutableArray`1 dynamicAnalysisSpans) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileSynthesizedMethods(TypeCompilationState compilationState) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamedType(NamedTypeSymbol containingType) [/src/CADLearning.Repository/CADLearning.Repository.csproj]
/usr/share/dotnet/sdk/2.2.202/Roslyn/Microsoft.CSharp.Core.targets(58,5): error :    at Microsoft.CodeAnalysis.CSharp.MethodCompiler.<>c__DisplayClass21_0.<CompileNamedTypeAsTask>b__0() [/src/CADLearning.Repository/CADLearning.Repository.csproj]`

**Comment by JohnGalt1717** [comment]
Only other thing I can think of is that this project references Mammoth which is a .net 4.6.1 nuget package that would warn in the past but would run/build fine.

**Comment by jaredpar** [comment]
@JohnGalt1717 thanks the stack trace help. Pretty sure that is a separate issue. Asked @cston to take a look at it.

**Comment by pkindruk** [comment]
@jaredpar same problem here, but completely different environment.
@JohnGalt1717 new roslyn compiler seems to be the case for my problem. What version do you fail with? And try finding code pattern methioned in this issue.

My brand new vs2019 failed to build my .NET Framework 4.5 class library saying `C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\Roslyn\Microsoft.CSharp.Core.targets(58,5): error MSB6006: "csc.exe" exited with code -2146232797.`
Also it gave me some errors in Windows logs
```
Faulting application name: csc.exe, version: 3.0.19.17001, time stamp: 0xb53a1b95
Faulting module name: unknown, version: 0.0.0.0, time stamp: 0x00000000
Exception code: 0x80131623
Fault offset: 0x00007ffc63c94e0c
Faulting process id: 0x29dc
Faulting application start time: 0x01d4ea31daf070eb
Faulting application path: C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\Roslyn\csc.exe
Faulting module path: unknown
Report Id: 0c5cf8ab-f582-4f0f-84d8-846a634d7ce9
Faulting package full name: 
Faulting package-relative application ID: 
```

```
Application: csc.exe
Framework Version: v4.0.30319
Description: The application requested process termination through System.Environment.FailFast(string message).
Message: System.InvalidOperationException: Unexpected value 'Beq' of type 'System.Reflection.Metadata.ILOpCode'
   at Microsoft.CodeAnalysis.CodeGen.ILOpCodeExtensions.GetLeaveOpcode(ILOpCode opcode)
   at Microsoft.CodeAnalysis.CodeGen.ILBuilder.BasicBlock.RewriteBranchesAcrossExceptionHandlers()
   at Microsoft.CodeAnalysis.CodeGen.ILBuilder.RewriteBranchesAcrossExceptionHandlers()
   at Microsoft.CodeAnalysis.CodeGen.ILBuilder.RealizeBlocks()
   at Microsoft.CodeAnalysis.CodeGen.ILBuilder.Realize()
   at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.GenerateImpl()
   at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.Generate(Int32& asyncCatchHandlerOffset, ImmutableArray`1& asyncYieldPoints, ImmutableArray`1& asyncResumePoints)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.GenerateMethodBody(PEModuleBuilder moduleBuilder, MethodSymbol method, Int32 methodOrdinal, BoundStatement block, ImmutableArray`1 lambdaDebugInfo, ImmutableArray`1 closureDebugInfo, StateMachineTypeSymbol stateMachineTypeOpt, VariableSlotAllocator variableSlotAllocatorOpt, DiagnosticBag diagnostics, DebugDocumentProvider debugDocumentProvider, ImportChain importChainOpt, Boolean emittingPdb, Boolean emitTestCoverageData, ImmutableArray`1 dynamicAnalysisSpans)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileSynthesizedMethods(TypeCompilationState compilationState)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamedType(NamedTypeSymbol containingType)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.<>c__DisplayClass21_0.<CompileNamedTypeAsTask>b__0()
Stack:
   at System.Environment.FailFast(System.String, System.Exception)
   at Microsoft.CodeAnalysis.FatalError.ReportUnlessCanceled(System.Exception)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler+<>c__DisplayClass21_0.<CompileNamedTypeAsTask>b__0()
   at Microsoft.CodeAnalysis.CodeGen.ILOpCodeExtensions.GetLeaveOpcode(System.Reflection.Metadata.ILOpCode)
   at Microsoft.CodeAnalysis.CodeGen.ILBuilder+BasicBlock.RewriteBranchesAcrossExceptionHandlers()
   at Microsoft.CodeAnalysis.CodeGen.ILBuilder.RewriteBranchesAcrossExceptionHandlers()
   at Microsoft.CodeAnalysis.CodeGen.ILBuilder.RealizeBlocks()
   at Microsoft.CodeAnalysis.CodeGen.ILBuilder.Realize()
   at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.GenerateImpl()
   at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator.Generate(Int32 ByRef, System.Collections.Immutable.ImmutableArray`1<Int32> ByRef, System.Collections.Immutable.ImmutableArray`1<Int32> ByRef)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.GenerateMethodBody(Microsoft.CodeAnalysis.CSharp.Emit.PEModuleBuilder, Microsoft.CodeAnalysis.CSharp.Symbols.MethodSymbol, Int32, Microsoft.CodeAnalysis.CSharp.BoundStatement, System.Collections.Immutable.ImmutableArray`1<Microsoft.CodeAnalysis.CodeGen.LambdaDebugInfo>, System.Collections.Immutable.ImmutableArray`1<Microsoft.CodeAnalysis.CodeGen.ClosureDebugInfo>, Microsoft.CodeAnalysis.CSharp.StateMachineTypeSymbol, Microsoft.CodeAnalysis.CodeGen.VariableSlotAllocator, Microsoft.CodeAnalysis.DiagnosticBag, Microsoft.CodeAnalysis.CodeGen.DebugDocumentProvider, Microsoft.CodeAnalysis.CSharp.ImportChain, Boolean, Boolean, System.Collections.Immutable.ImmutableArray`1<Microsoft.CodeAnalysis.CodeGen.SourceSpan>)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileSynthesizedMethods(Microsoft.CodeAnalysis.CSharp.TypeCompilationState)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamedType(Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol)
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler+<>c__DisplayClass21_0.<CompileNamedTypeAsTask>b__0()
   at Roslyn.Utilities.UICultureUtilities+<>c__DisplayClass5_0.<WithCurrentUICulture>b__0()
   at System.Threading.Tasks.Task.Execute()
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef)
   at System.Threading.Tasks.Task.ExecuteEntry(Boolean)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
```
Then I opened my sandbox console app (.NET Framework 4.6.1) and used code issue starter provided. Same result. I went to the project originally failing and found code like this(I omitted all business logic):
```csharp
class A
{
    private async Task DoSmth()
    {
        try
        {
            try
            {
                //1
                return;
            }
            finally
            {
                //2
                await Task.CompletedTask;
            }
        }
        catch { }
        finally
        {
            //3
            await Task.CompletedTask;
        }
    }
}
```
Removing any of numbered lines makes code compile. 

My primary project is still compiling with vs2017. So I put that simplified class above in the console app right next to plain `Main` method and tried to compile on vs2017 (success) and vs2019 (failure). All projects were created under vs2017 and opened in vs2019 if that matters.

And last thing csc.exe versions:
```
c:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Roslyn>csc -version
2.10.0.0 (b9fb1610)

c:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\Roslyn>csc -version
3.0.19.17001 (1deafee3)
```

**Comment by gingters** [comment]
Hello.
I also see that issue. It compiles fine in VS 2017 and not in VS 2019.

This is our solution: https://github.com/thinktecture/relayserver/tree/2.2.0
In the project `OnPremiseConnector` in the class `RelayServerConnection` we have a try -> try -> finally -> await constellation that might be another version of #34720.

This issue is blocking updating to VS 2019.

**Comment by cston** [comment]
@gingters, #34791 should fix the code gen issue with an await in a nested finally clause. In the meantime, before the fix is available, are you able to get unblocked by extracting the nested try/finally to a separate method or local function?

**Comment by springy76** [comment]
Starting with **VS2019 PREVIEW16.3.0 Preview X** (1..4) I get this exact message for one (large!) project in a solution containing 132 projects. It's the second-important project in this solution :(

Using **VS2019 RELEASE 16.2.X** everything is ok. I fear when 16.3 will get released this bug still will be present. What can I do to narrow this problem down?

**Comment by cston** [comment]
@springy76, thanks for reporting the latest issue. Is there a stack trace for the failure in `csc.exe` in the Windows event log? If so, please share the stack trace if you're comfortable doing so. Thanks.

**Comment by springy76** [comment]
@cston I found the event log later and then created a new issue #38725
